### PR TITLE
[nrk:series] fix and update extractor

### PR DESCRIPTION
**Pull request by Roxedus**
Added new error message
Fixed tests, some of the existing ones were removed from the site

## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fixes a bug i noticed when trying to fetch a full series, rather than each individual episode.

NRK introduced malformed json to their site, creating values like `"subtitle":undefined`, using `.replace("undefined", "'undefined'")` as i fail to see a function in utils.py to solve this.

I also updated the tests to not use content which has expired, and which should not expire, at least for a while. Also added the error message expired items gives.

### CODE REVIEW

File: youtube_dl/extractor/nrk.py [hunk 1]
```diff
@@ -408,7 +440,7 @@ def _extract_series(self, webpage, display_id, fatal=True):
             self._search_regex(
                 (r'INITIAL_DATA(?:_V\d)?_*\s*=\s*({.+?})\s*;',
                  r'({.+?})\s*,\s*"[^"]+"\s*\)\s*</script>'),
-                webpage, 'config', default='{}' if not fatal else NO_DEFAULT),
+                webpage.replace("undefined", '"undefined"'), 'config', default='{}' if not fatal else NO_DEFAULT),
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: Read coding conventions.

Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: `js_to_json`.

File: youtube_dl/extractor/nrk.py [hunk 2]
```diff
@@ -479,6 +511,16 @@ class NRKTVSeriesIE(NRKTVSerieBaseIE):
     _VALID_URL = r'https?://(?:tv|radio)\.nrk(?:super)?\.no/serie/(?P<id>[^/]+)'
     _ITEM_RE = r'(?:data-season=["\']|id=["\']season-)(?P<id>\d+)'
     _TESTS = [{
+        'url': 'https://tv.nrk.no/serie/blank',
+        'info_dict': {
+            'id': 'blank',
+            'title': 'Blank',
+            'description': 'Da Markus dro fra bygda var han den kuleste fyren med den fineste dama. '
+                           'Etter kort tid er han tilbake. Og ingenting er som før. Norsk dramaserie.',
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: Read coding conventions.

File: youtube_dl/extractor/nrk.py [hunk 3]
```diff
@@ -370,7 +383,27 @@ class NRKTVIE(NRKBaseIE):
...
-    _TEST = {
+    _TESTS = [{
+        'url': 'https://tv.nrk.no/serie/hellums-kro/sesong/1/episode/2',
+        'info_dict': {
+            'id': 'MUHH36005220BA',
+            'ext': 'mp4',
+            'title': 'Kro, krig og kjærlighet 2:6',
+            'description': 'Jan tviholder på sine prinsipper. '
+                           'Oddbjørn er skeptisk til Jans evne til å lede '
+                           'dem og bestemmer seg for å ta saken i egne hender.',
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: ^

File: youtube_dl/extractor/nrk.py [hunk 4]
```diff
@@ -479,6 +513,15 @@ class NRKTVSeriesIE(NRKTVSerieBaseIE):
     _VALID_URL = r'https?://(?:tv|radio)\.nrk(?:super)?\.no/serie/(?P<id>[^/]+)'
     _ITEM_RE = r'(?:data-season=["\']|id=["\']season-)(?P<id>\d+)'
     _TESTS = [{
+        'url': 'https://tv.nrk.no/serie/blank',
+        'info_dict': {
+            'id': 'blank',
+            'title': 'Blank',
+            'description': 'Da Markus dro fra bygda var han den kuleste fyren med den fineste dama. Etter kort tid er han tilbake. Og ingenting er som før. Norsk dramaserie.',
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: ^

File: youtube_dl/extractor/nrk.py [hunk 5]
```diff
@@ -16,6 +16,7 @@
     parse_age_limit,
     parse_duration,
     try_get,
+    js_to_json,
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: Alphabetic.

File: youtube_dl/extractor/nrk.py (hunk 4 above)
Comment (Roxedus) [reply youtube_dl/extractor/nrk.py]: Is there something wrong with this line, or is it to show what to do on the line above?

Comment (dstftw) [reply youtube_dl/extractor/nrk.py]: Same as in previous review.

File: youtube_dl/extractor/nrk.py [hunk 6]
```diff
@@ -405,10 +436,11 @@ def _real_extract(self, url):
 class NRKTVSerieBaseIE(InfoExtractor):
     def _extract_series(self, webpage, display_id, fatal=True):
         config = self._parse_json(
-            self._search_regex(
-                (r'INITIAL_DATA(?:_V\d)?_*\s*=\s*({.+?})\s*;',
-                 r'({.+?})\s*,\s*"[^"]+"\s*\)\s*</script>'),
-                webpage, 'config', default='{}' if not fatal else NO_DEFAULT),
+            js_to_json(
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: In `transform_source=`.

File: youtube_dl/extractor/nrk.py (hunk 4 above)
Comment (dstftw) [reply youtube_dl/extractor/nrk.py]: Nothing changed.

File: youtube_dl/extractor/nrk.py [hunk 7]
```diff
@@ -105,6 +106,7 @@ def video_id_and_title(idx):
             MESSAGES = {
                 'ProgramRightsAreNotReady': 'Du kan dessverre ikke se eller høre programmet',
                 'ProgramRightsHasExpired': 'Programmet har gått ut',
+                'NoProgramRights': 'Ikke tilgjengelig',
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: Provide example URL.

File: youtube_dl/extractor/nrk.py [hunk 8]
```diff
@@ -370,7 +383,25 @@ class NRKTVIE(NRKBaseIE):
...
 class NRKTVEpisodeIE(InfoExtractor):
     _VALID_URL = r'https?://tv\.nrk\.no/serie/(?P<id>[^/]+/sesong/\d+/episode/\d+)'
-    _TEST = {
+    _TESTS = [{
+        'url': 'https://tv.nrk.no/serie/hellums-kro/sesong/1/episode/2',
+        'info_dict': {
+            'id': 'MUHH36005220BA',
+            'ext': 'mp4',
+            'title': 'Kro, krig og kjærlighet 2:6',
+            'description': 'Jan tviholder på sine prinsipper. Oddbjørn er skeptisk til Jans evne til å lede dem og bestemmer seg for å ta saken i egne hender.',
```
Comment (dstftw) [inline youtube_dl/extractor/nrk.py]: ^

File: youtube_dl/extractor/nrk.py (hunk 4 above)
Comment (Roxedus) [reply youtube_dl/extractor/nrk.py]: I clearly don't understand whats missing here.

File: youtube_dl/extractor/nrk.py (hunk 7 above)
Comment (Roxedus) [reply youtube_dl/extractor/nrk.py]: Do you want it in the code, or in this discussion?

Comment (dstftw) [reply youtube_dl/extractor/nrk.py]: Add as a test.

File: youtube_dl/extractor/nrk.py (hunk 4 above)
Comment (dstftw) [reply youtube_dl/extractor/nrk.py]: Hide it behind md5 as you did with the first test. What's not clear?

File: youtube_dl/extractor/nrk.py (hunk 7 above)
Comment (Roxedus) [reply youtube_dl/extractor/nrk.py]: A valid url for the message is already in a test, second on NRKTV

File: youtube_dl/extractor/nrk.py (hunk 4 above)
Comment (Roxedus) [reply youtube_dl/extractor/nrk.py]: That really didn't click in my head until you pointed it out, god i feel stupid

**Pull request merged**